com_migratetojoomla/admin/src/Helper/DownloadHelper.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_migratetojoomla
+ *
+ * @copyright   (C) 2006 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+
+namespace Joomla\Component\MigrateToJoomla\Administrator\Helper;
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\Language\Text;
+
+use Joomla\Component\MigrateToJoomla\Administrator\Helper\HttpHelper;
+use Joomla\Component\MigrateToJoomla\Administrator\Helper\FilesystemHelper;
+use Joomla\Component\MigrateToJoomla\Administrator\Helper\FtpHelper;
+use Joomla\Component\MigrateToJoomla\Administrator\Helper\MainHelper;
+// phpcs:disable PSR1.Files.SideEffects
+\defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+
+class DownloadHelper
+{
+    public  static $downloadmanager;
+    /**
+     * Method to check connection with respective method
+     * 
+     * @param array form data
+     * @return boolean True on success
+     * 
+     * since
+     */
+    public static function testconnection($data = [])
+    {
+        $method = $data['mediaoptions'];
+
+        if ($method == 1) {
+            // Http
+            HttpHelper::testconnection($data['livewebsiteurl']);
+        } else if ($method == 2) {
+            // File system
+            FilesystemHelper::testconnection($data['basedir']);
+        }
+    }
+
+    /**
+     * Method to Download 
+     * 
+     * @param array form data
+     * 
+     */
+    public static function download($data = [])
+    {
+        $method = $data['mediaoptions'];
+
+        switch ($method) {
+            case 2:
+                DownloadHelper::$downloadmanager = new FilesystemHelper;
+                break;
+            case 3:
+                DownloadHelper::$downloadmanager = new FtpHelper;
+                break;
+            case 1:
+            default:
+                DownloadHelper::$downloadmanager = new HttpHelper;
+                break;
+        }
+
+        $source = MainHelper::addtrailingslashit($data['basedir']) . 'wp-content\uploads\\';
+
+        $destination = MainHelper::addtrailingslashit(JPATH_ROOT) . 'images\\';
+        $app   = Factory::getApplication();
```

Comment:
Make sure that when 2 variables are set, one after the other without a blank line in between, that you write:
$destination = ...
$app             = ...